### ISSUE TITLE
make:auth Add supports method to generated LoginFormAuthenticator

### ISSUE THREAD

**Issue by ahundiak** [symfony/maker-bundle#1056]
There has been a number of issues reported with the default AbstractLoginFormAuthenticator::supports method.  It compares actual urls and often fails based on web server configuration. There is a bug report being discussed but the problem could be solved by adding a supports method to the generated 5.2+ LoginFormAuthenticator template.
```
    public function supports(Request $request)
    {
        return self::LOGIN_ROUTE === $request->attributes->get('_route')
            && $request->isMethod('POST');
    }
```
It's the exact same code that was used in the pre-5.2 guard authenticator.  I don't recall ever encountering an issue.  It's not really appropriate to use _route in a component but I don't see any problem with using it at the bundle level.   

Seems like a simple enough fix that would solve quite a few "I can't login and there are no errors showing" problems.  I could submit a PR.

**Comment by JB-oclock** [comment]
i'm not sure the problem should be fixed in the make:auth.

As i understand it, AbstractLoginFormAuthenticator implements a default way to handle all POST request to the login path.

https://github.com/symfony/security-http/blob/3d4b612da3a278285e6fd16fc2e5233820eeba0d/Authenticator/AbstractLoginFormAuthenticator.php#L42-L45

Maybe adding just something to show that's here you should add your code to implements your logic

```
    public function supports(Request $request): bool
    {
      // here you can override default handle to your context (web server, path ...)
      // you MUST check if the method is POST 
      // $request->isMethod('POST') 
      // AND 
      // $this->getLoginUrl($request) // with apache will be something like /some/directory/public/login
      // $request->getPathInfo() // will be something like /login
      // for exemple it could be 
      //return $request->isMethod('POST') && '/login' === $request->getPathInfo();
    }
```

**Comment by ahundiak** [comment]
The more I look at it the more I am convinced that AbstractLoginFortAuthenticator::suports is just plain broke.

There is a Http/Firewall/LogoutListener which is really a kernel.request listener that listens for /logout.  It's implemented to sort of look like an authenticator with a supports and authenticate method.

It's supports method look like:
```
    public function supports(Request $request): ?bool
    {
        return $this->requiresLogout($request);
    }
    protected function requiresLogout(Request $request): bool
    {
        return isset($this->options['logout_path']) && $this->httpUtils->checkRequestPath($request, $this->options['logout_path']);
    }
```
If you then look at checkRequestPath you see a bunch of code designed to look for /logout. 
```
public function checkRequestPath(Request $request, string $path): bool
    {
        if ('/' !== $path[0]) {
            try {
                // matching a request is more powerful than matching a URL path + context, so try that first
                if ($this->urlMatcher instanceof RequestMatcherInterface) {
                    $parameters = $this->urlMatcher->matchRequest($request);
                } else {
                    $parameters = $this->urlMatcher->match($request->getPathInfo());
                }

                return isset($parameters['_route']) && $path === $parameters['_route'];
            } catch (MethodNotAllowedException $e) {
                return false;
            } catch (ResourceNotFoundException $e) {
                return false;
            }
        }

        return $path === rawurldecode($request->getPathInfo());
    }
```

The bottom line is that the simplistic check in AbstractLogin is not enough to cover all the use cases.  Maybe I am missing something but I see no reason the check for /login should be different then the check for /logout.

Tweaking the maker bundle avoids these issues.  Fixing the authenticator is probably better.

On the other hand, it's been a few months since I seen any stackoverflow questions on this.  Maybe people are finding the existing answer or maybe the problem was solved in some other fashion.

**Comment by jrushlow** [comment]
Closing - See https://github.com/symfony/maker-bundle/pull/1523 

`make:auth` is being deprecated - new features should be implemented in one of the `make:security:*` commands

We're are also recommending usage of `make:security:form-login` which uses the Symfony Security components - [FormLoginAuthenticator](https://github.com/symfony/symfony/blob/7.0/src/Symfony/Component/Security/Http/Authenticator/FormLoginAuthenticator.php)
